crates/shared/src/price_estimation/buffered.rs

```diff
@@ -119,7 +119,7 @@ impl PriceEstimating for BufferingPriceEstimator {
             in_flight_requests.extend(
                 new_requests
                     .iter()
-                    .map(|(query, fut)| (*query, fut.downgrade().unwrap())),
+                    .map(|(query, fut)| (*query, fut.downgrade().expect("future completed"))),
```

Comment:
Unrelated to this PR. Just something which slipped through the cracks in the previous PR.

Reply:
Another way of doing it is keep unwrap with a comment on top of it. Personally I don't find the expect messages useful because in a panic you already get the line number so you know where it happened and can check it in the code. Comments are a lot easier to write and format than the expect string so I like them more.

Reply:
I'm OK with either a comment or `expect`.